### PR TITLE
Stop sync if change network and genesis info not match

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

File: packages/neuron-wallet/src/startup/sync-block-task/index.ts [hunk 1]
```diff
@@ -46,6 +46,13 @@ const networkChange = async (network: NetworkWithID) => {
     DatabaseInitSubject.getSubject().next(databaseInitParams)
     // re init txCount in addresses if switch network
     await updateAllAddressesTxCount(network.remote)
+  } else {
+    // if killed, should stop previous one
+    DatabaseInitSubject.getSubject().next({
+      network,
+      genesisBlockHash: 'stop',
```
Comment (ashchan) [inline packages/neuron-wallet/src/startup/sync-block-task/index.ts]: Is an empty string even better?

File: packages/neuron-wallet/src/startup/sync-block-task/task.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ const testIndexer = async (url: string): Promise<boolean> => {
 const run = async () => {
   DatabaseInitSubject.getSubject().subscribe(async (params: DatabaseInitParams) => {
     const { network, genesisBlockHash, chain } = params
-    if (network && genesisBlockHash.startsWith('0x')) {
+    if (network && (genesisBlockHash.startsWith('0x') || genesisBlockHash === 'stop')) {
```
Comment (ashchan) [inline packages/neuron-wallet/src/startup/sync-block-task/task.ts]: Why it should go through to start the queue when `stopped`?

File: packages/neuron-wallet/src/startup/sync-block-task/index.ts (hunk 1 above)
Comment (classicalliu) [reply packages/neuron-wallet/src/startup/sync-block-task/index.ts]: Means '0x' here ?

File: packages/neuron-wallet/src/startup/sync-block-task/task.ts (hunk 2 above)
Comment (classicalliu) [reply packages/neuron-wallet/src/startup/sync-block-task/task.ts]: The stop queue logic is in ` syncSwitchNetwork`.

**Comment by ashchan** [comment]
Let's leave it as. My refactoring on sync would mostly wipe these out ;P

**Comment by ashchan** [comment]
Closing this as we're having dramatic changes to this module.

**Pull request closed without merging**